### ISSUE TITLE
Retrain tutorial is not working

### ISSUE THREAD

**Issue by neilyoung** [google-coral/edgetpu#330]
Hi,

I can't get anything useful from the retrain tutorial https://coral.ai/docs/edgetpu/retrain-detection/#configure-your-own-training-data.

I come through (even though there are errors in the tutorial) but the dog remains undetected.

If I try to change NUM_EVAL_STEPS and retrain I'm catching this in the docker container:

```
WARNING:tensorflow:Forced number of epochs for all eval validations to be 1.
WARNING:tensorflow:Expected number of evaluation epochs is 1, but instead encountered `eval_on_train_input_config.num_epochs` = 0. Overwriting `num_epochs` to 1.
WARNING:tensorflow:Estimator's model_fn (<function model_fn at 0x7fcefb711aa0>) includes params argument, but params are not passed to Estimator.


```

Do I have to clean some directories first before restarting?

**Comment by neilyoung** [comment]
I removed the `~/google-coral/tutorials/docker/object_detection/out/train` dir and it works as it seems

**Comment by neilyoung** [comment]
Arghh... this entire retrain stuff is a big mess....

After having changed NUM_EVAL_STEPS from 100 to 200 it runs for hours and hours just to stop like so:

```
DONE (t=0.33s).
Accumulating evaluation results...
DONE (t=0.06s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.703
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.954
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.843
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.468
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.743
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.748
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.756
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.756
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.508
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.794
WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow/python/estimator/estimator.py:1044: calling add_meta_graph_and_variables (from tensorflow.python.saved_model.builder_impl) with legacy_init_op is deprecated and will be removed in a future version.
Instructions for updating:
Pass your op to the equivalent parameter main_op instead.
WARNING:tensorflow:From /usr/local/lib/python2.7/dist-packages/tensorflow/python/estimator/estimator.py:1044: calling add_meta_graph_and_variables (from tensorflow.python.saved_model.builder_impl) with legacy_init_op is deprecated and will be removed in a future version.
Instructions for updating:
Pass your op to the equivalent parameter main_op instead.


```

**Comment by Namburger** [comment]
@neilyoung Are you sure the process finished? 
It seems like normal logs at this point to me.

The full documentation for Object Detection API is here: https://tensorflow-object-detection-api-tutorial.readthedocs.io/en/tensorflow-1.14/
Our tutorial basically wraps around that

**Comment by neilyoung** [comment]
Yes. It’s back at the prompt after that. Happened twice

**Comment by neilyoung** [comment]
Wow, that's strange. I was searching for the warning message, which preceeded the break, and found this: https://github.com/tensorflow/models/issues/5707

Blindly followed the instruction to add the log level info line at the top of the script. Output was verbose, but at least it came through for the very first time with NUM_STEPS...

Let's see if it detects something now.

**Comment by neilyoung** [comment]
> The full documentation for Object Detection API is here: https://tensorflow-object-detection-api-tutorial.readthedocs.io/en/tensorflow-1.14/
Our tutorial basically wraps around that

Thanks for the link. Well, this tutorial talks about Python 3.7. What I see from your scripts is that it used Python 2.7

**Comment by neilyoung** [comment]
But even with NUM_EVAL_STEPS=200 it does not produce any result, so this is the output.

```
decades@ubuntu:~/google-coral/tflite/python/examples/detection$ python3 detect_image.py   --model ${HOME}/google-coral/tutorials/docker/object_detection/out/models/ssd_mobilenet_v1_catsdogs_quant_edgetpu.tflite   --labels ${HOME}/google-coral/tutorials/docker/object_detection/out/models/labels.txt   --input ${HOME}/google-coral/tutorials/docker/object_detection/out/models/dog.jpg   --output dog_result.jpg
----INFERENCE TIME----
Note: The first inference is slow because it includes loading the model into Edge TPU memory.
144.75 ms
17.58 ms
17.63 ms
17.22 ms
16.47 ms
-------RESULTS--------
No objects detected

```

Doesn't work for me. 

BTW: As with your private project also this tutorial has a problem with the final path parameter. It uses "detection" instead of "object_detection". Where would I be able to fix at least this?

**Comment by Namburger** [comment]
@neilyoung 
have you tried it with different images? With just 200 steps, I wouldn't expect much accuracy..

**Comment by neilyoung** [comment]
No, I'm a real beginner. I was just blindly following the tutorial.

What I did so far

NUM_TRAINING_STEPS=500 && NUM_EVAL_STEPS=100: Came through, no detection
NUM_TRAINING_STEPS=500 && NUM_EVAL_STEPS=200: Came through one out of three, no detection
NUM_TRAINING_STEPS=1000 && NUM_EVAL_STEPS=500: Aborted somewhen after hours
NUM_TRAINING_STEPS=5000 && NUM_EVAL_STEPS=1000: Crashed very quickly ("Killed").

It is an Asus i5, 16 GB, no GPU, Ubuntu 18.04. 

It just sucks.

**Comment by Namburger** [comment]
“Killed” is a signal from the OOM killer... nothing we can do about that, you’ll need more ram, unfortunately
